develop: TestWindow_DragObjectThatMoves fail

This was extracted out of #1814 because the failures have different origins.

When running the GLFW window_test.go, TestWindow_DragObjectThatMoves fails.

This failure was
- originally introduced by e5bfd3c4d53ba6c770664c7bdbd5181fa02dc543 of #1783
- modified by bb4c70a554d62478e6897e8fbc2368eb47c3d834 and 4759e3a5f2a32f10baf519e556853d4503ba0e3a of #1789

The current outcome is:

```
=== RUN   TestWindow_DragObjectThatMoves
    window_test.go:392: 
        	Error Trace:	window_test.go:392
        	Error:      	Not equal: 
        	            	expected: &fyne.DragEvent{PointEvent:fyne.PointEvent{AbsolutePosition:fyne.Position{X:10, Y:10}, Position:fyne.Position{X:7, Y:7}}, Dragged:fyne.Delta{DX:2, DY:2}}
        	            	actual  : &fyne.DragEvent{PointEvent:fyne.PointEvent{AbsolutePosition:fyne.Position{X:10, Y:10}, Position:fyne.Position{X:6, Y:6}}, Dragged:fyne.Delta{DX:2, DY:2}}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -7,4 +7,4 @@
        	            	   Position: (fyne.Position) {
        	            	-   X: (float32) 7,
        	            	-   Y: (float32) 7
        	            	+   X: (float32) 6,
        	            	+   Y: (float32) 6
        	            	   }
        	Test:       	TestWindow_DragObjectThatMoves
--- FAIL: TestWindow_DragObjectThatMoves (0.21s)
```

**Note**: it might be necessary to disable the loop’s window closed handling (lines 103-148 of glfw/loop.go) to ensure that the test runs do not exit prematurely.

--- a/internal/driver/glfw/window.go
+++ b/internal/driver/glfw/window.go
@@ -649,9 +649,10 @@ func (w *window) mouseMoved(viewport *glfw.Window, xpos float64, ypos float64) {
 
 	if w.mouseDragged != nil {
 		if w.mouseButton > 0 {
+			draggedObjPos := w.mouseDragged.(fyne.CanvasObject).Position()
 			ev := new(fyne.DragEvent)
 			ev.AbsolutePosition = w.mousePos
-			ev.Position = w.mousePos.Subtract(w.mouseDraggedOffset)
+			ev.Position = w.mousePos.Subtract(w.mouseDraggedOffset).Subtract(draggedObjPos)
 			ev.Dragged = fyne.NewDelta(w.mousePos.X-w.mouseDragPos.X, w.mousePos.Y-w.mouseDragPos.Y)
 			wd := w.mouseDragged
 			w.queueEvent(func() { wd.Dragged(ev) })

